File: Helpers/src/main/java/com/dg/helpers/DoubleHelper.java
package com.dg.helpers;

/**
 * Created by Daniel Cohen Gindi (danielgindi@gmail.com)
 */
public class DoubleHelper
{
    public static Double withObject(Object value)
    {
        if (value == null) return null;
        else if (value instanceof Double) return (Double)value;
        else if (value instanceof Float) return (double) (Float) value;
        else if (value instanceof Integer) return (double) (Integer) value;
        else if (value instanceof Short) return (double) (Short) value;
        else return null;
    }

    public static float[] toArray(Float[] inArray)
    {
        float[] array = new float[inArray.length];
        for (int i = 0, len = inArray.length; i < len; i++)
        {
            Float value = inArray[i];

            if (value != null)
            {
                array[i] = value;
            }
        }
        return array;
    }
}

File: Helpers/src/main/java/com/dg/helpers/DoubleHelper.kt
package com.dg.helpers

@Suppress("unused")
object DoubleHelper
{
    @Suppress("MemberVisibilityCanBePrivate")
    fun withObject(value: Any?): Double?
    {
        when (value)
        {
            null -> return null
            is Double -> return value
            is Float -> return value.toDouble()
            is Int -> return value.toDouble()
            is Short -> return value.toDouble()
            is String -> try
            {
                return java.lang.Double.parseDouble(value)
            }
            catch (ignored: Exception)
            {
            }
        }
        return null
    }

    @Suppress("MemberVisibilityCanBePrivate")
    fun withObject(value: Any?, defaultValue: Double): Double
    {
        return withObject(value) ?: defaultValue
    }

    fun toArray(inArray: Array<Float?>): FloatArray
    {
        val array = FloatArray(inArray.size)
        var i = 0
        val len = inArray.size
        while (i < len)
        {
            val value = inArray[i]

            if (value != null)
            {
                array[i] = value
            }
            i++
        }
        return array
    }
}
